### PR TITLE
Refresh token

### PR DESCRIPTION
**Pull request by emir4ik-code**

### CODE REVIEW

File: src/services/auth.service.ts [hunk 1]
```diff
@@ -14,29 +14,44 @@ class AuthService extends Repository<UserEntity> {
     super();
   }
 
-  private createToken(data: BaseUserResponseDTO): TokenData {
-    const time = 3600 * 100;
-    const expiresIn = String(time);
-    return { expiresIn, token: sign({ ...data }, SECRET_KEY, { expiresIn }) };
+  public createToken(data: BaseUserResponseDTO, type: 'access' | 'refresh'): TokenData {
+    if (type === 'access') {
```
Comment (raylu511) [inline src/services/auth.service.ts:18]: nit: you can do 
```
const [expiresIn, secretKey] = (() => {
  if (type === 'access') {
    return ['600000', ACCESS_TOKEN_SECRET_KEY]
  }
  return ['30 days', REFRESH_TOKEN_SECRET_KEY]
})();
 return { expiresIn, token: sign({ ...data }, secretKey, { expiresIn }) };
```

File: src/middlewares/auth.middleware.ts [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
-      next(new HttpException(404, 'Authentication token missing'));
-    }
+  try {
+    const { id } = (await verify(refreshToken, REFRESH_TOKEN_SECRET_KEY)) as DataStoredInToken;
+    const userInfo = await UserEntity.findOne({
+      where: { id },
+      select: ['id', 'email', 'firstName', 'lastName', 'dateOfBirth', 'language', 'profilePic', 'updatedAt', 'createdAt'],
+    });
+    req.user = userInfo;
+    /* jscpd:ignore-end */
```
Comment (raylu511) [inline src/middlewares/auth.middleware.ts:36]: what's this for?

File: src/services/auth.service.ts [hunk 3]
```diff
@@ -14,29 +14,44 @@ class AuthService extends Repository<UserEntity> {
...
+    } else {
+      const expiresIn = '30 days';
+      return { expiresIn, token: sign({ ...data }, REFRESH_TOKEN_SECRET_KEY, { expiresIn }) };
+    }
   }
 
-  private createCookie(tokenData: TokenData): string {
-    return `Authorization=${tokenData.token}; HttpOnly; Max-Age=${tokenData.expiresIn};`;
+  public createCookie(tokenData: TokenData, type: 'access' | 'refresh'): string {
+    if (type === 'access') return `AccessToken=${tokenData.token}; HttpOnly; Max-Age=${tokenData.expiresIn};`;
```
Comment (raylu511) [inline src/services/auth.service.ts:28]: I would do something like 
```
const tokenType = type === 'access' ? 'AccessToken' : 'RefreshToken';
return `${tokenType}=${tokenData.token}; HttpOnly; Max-Age=${tokenData.expiresIn};`;
```

File: src/middlewares/auth.middleware.ts [hunk 4]
```diff
@@ -1,31 +1,46 @@
...
-import { SECRET_KEY } from '@/config';
+import { ACCESS_TOKEN_SECRET_KEY, REFRESH_TOKEN_SECRET_KEY } from '@/config';
 import { UserEntity } from '@/entities';
 import { HttpException } from '@/exceptions';
+import { AuthService } from '@/services';
 import { DataStoredInToken, RequestWithUser } from '@/interfaces';
 
-const authMiddleware = async (req: RequestWithUser, _res: Response, next: NextFunction) => {
+const authMiddleware = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+  const refreshToken = req.cookies['RefreshToken'] || null;
```
Comment (raylu511) [inline src/middlewares/auth.middleware.ts:10]: Wouldn't this default to a falsy value if it doesn't exist in the cookie?
If it is, adding null is unneccessary.

File: src/services/auth.service.ts (hunk 1 above)
Comment (emir4ik-code) [reply src/services/auth.service.ts:18]: Neat!

File: src/controllers/auth.controller.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ class AuthController {
     try {
       const userData: LoginUserRequestDto = req.body;
       const { accessTokenCookie, refreshTokenCookie, data } = await this.authService.login(userData);
+      console.log(accessTokenCookie, refreshTokenCookie)
```
Comment (DLiSWE) [inline src/controllers/auth.controller.ts:23]: remove console.log

File: src/services/auth.service.ts [hunk 6]
```diff
@@ -14,29 +14,42 @@ class AuthService extends Repository<UserEntity> {
...
+        return ['600000', ACCESS_TOKEN_SECRET_KEY];
+      }
+      return ['30 days', REFRESH_TOKEN_SECRET_KEY];
+    })();
+    return { expiresIn, token: sign({ ...data }, secretKey, { expiresIn }) };
   }
 
-  private createCookie(tokenData: TokenData): string {
-    return `Authorization=${tokenData.token}; HttpOnly; Max-Age=${tokenData.expiresIn};`;
+  public createCookie(tokenData: TokenData, type: 'access' | 'refresh'): string {
```
Comment (raylu511) [inline src/services/auth.service.ts:27]: nit: you can destructure the properties of tokenData

File: src/controllers/auth.controller.ts (hunk 5 above)
Comment (emir4ik-code) [reply src/controllers/auth.controller.ts]: done

File: src/services/auth.service.ts (hunk 6 above)
Comment (emir4ik-code) [reply src/services/auth.service.ts:27]: done

**Review by DLiSWE — COMMENTED** [review]
LGTM+

**Review by DLiSWE — APPROVED** [review]

**Review by raylu511 — APPROVED** [review]

**Pull request merged**